### PR TITLE
feat: Allow additional tools for sphinx_docs

### PR DESCRIPTION
**Pull request by castler**
Some Sphinx plugins require that other (not necessarily Python) tools are available.

One example is the plugin https://github.com/basejumpa/sphinxcontrib-umlet, which requires that UMLet is somehow within the sandbox.

* Adds `tools` arg to `sphinx_docs` to allow passing in arbitrary tools
  that are made available at runtime
* Performs location expansion on `extra_opts`, which allows passing the
  location of the tools onto sphinx.

### CODE REVIEW

**Comment by aignas** [comment]
Since sphinx is enabled in the next release Ithink this does not need an extrachngalog entry, but the docstring for the tools could include a short example with a particular plugin. Why the opts need expansion is still a little unclear to me. Is it that some plugins may expect the tools to be found in the PATH and that case would not be covered. To be honest, I am not super familiar with sphinx and these questions could be just my ignorance.

**Comment by castler** [comment]
> . Why the opts need expansion is still a little unclear to me. Is it that some plugins may expect the tools to be found in the PATH and that case would not be covered.

Sure, something like this?

```
sphinx_docs(
    ...
    name = ...,
    srcs = ...,
    extra_opts = [
        "-Dsome_binary=$(execpath //:some_binary_target)",
    ],
    tools = [
        "//:some_binary_target",
    ],
)
```

File: sphinxdocs/private/sphinx.bzl [hunk 1]
```diff
@@ -238,7 +245,8 @@ def _run_sphinx(ctx, format, source_path, inputs, output_prefix):
     args.add("-j", "auto")  # Build in parallel, if possible
     args.add("-E")  # Don't try to use cache files. Bazel can't make use of them.
     args.add("-a")  # Write all files; don't try to detect "changed" files
-    args.add_all(ctx.attr.extra_opts)
+    for opt in ctx.attr.extra_opts:
+        args.add(ctx.expand_location(opt))
```
Comment (rickeylev) [inline sphinxdocs/private/sphinx.bzl:249]: Please update the doc to note that extra_opts performs location expansion

File: sphinxdocs/private/sphinx.bzl [hunk 2]
```diff
@@ -88,6 +89,7 @@ def sphinx_docs(
             source files. e.g., given `//docs:foo.md`, stripping `docs/`
             makes Sphinx see `foo.md` in its generated source directory.
         extra_opts: (list[str]) Additional options to pass onto Sphinx building.
+        tools: (list[label]) Additional tools that are used by Sphinx and its plugins.
```
Comment (rickeylev) [inline sphinxdocs/private/sphinx.bzl:92]:
```suggestion
        tools: (list[label]) Additional tools that are used by Sphinx and its plugins.
            This just makes the tools available during Sphinx execution. To locate
            them, use `extra_opts` and `$(location)` expansion
```

**Review by rickeylev — COMMENTED** [review]
Mostly lgtm, just need the extra_opts doc updated (I couldn't suggest an edit since it wasn't a modified line)

**Comment by castler** [comment]
> Mostly lgtm, just need the extra_opts doc updated (I couldn't suggest an edit since it wasn't a modified line)

Thanks, applied requested changes.

**Review by rickeylev — APPROVED** [review]

**Pull request merged**